From 3d431e9668aaa9d4aef0d7832c41c30af7e0b558 Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Thu, 3 Mar 2022 11:18:19 -0800
Subject: mitigate flaky tests temporarily (#37672)

* mitigate flaky tests temporarily

TestClusterLocal has a known fix which will land within a week.

The other one does not; we are not skipping though just making the test
slightly less exhaustive

* Update multicluster_test.go
---
 tests/integration/security/util/reachability/context.go | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tests/integration/security/util/reachability/context.go b/tests/integration/security/util/reachability/context.go
index 228a01c07f..5645c1437a 100644
--- a/tests/integration/security/util/reachability/context.go
+++ b/tests/integration/security/util/reachability/context.go
@@ -198,7 +198,11 @@ func Run(testCases []TestCase, t framework.TestContext, apps *util.EchoDeploymen
 												t.Skip("https://github.com/istio/istio/issues/37307")
 											}
 
-											src.CallWithRetryOrFail(t, opts)
+											retryOpts := []retry.Option{}
+											retryOpts = append(retryOpts, echo.DefaultCallRetryOptions()...)
+											retryOpts = append(retryOpts, retry.Converge(1)) // TODO(https://github.com/istio/istio/issues/37629) go back to converge
+
+											src.CallWithRetryOrFail(t, opts, retryOpts...)
 										})
 								}
 							}
-- 
2.35.3

